web/src/components/tasks/anytime-action/tax-clearance-certificate/AnytimeActionTaxClearanceCertificateElement.test.tsx

```diff
@@ -123,6 +133,128 @@ describe("<AnyTimeActionTaxClearanceCertificateReviewElement />", () => {
+      selectDropdownValueByText("Address state", "MD", false);
+      fillText("Address zip code", "21210");
+      fillText("Tax id", "012345678901");
+      fillText("Tax pin", "1234");
+
+      expect(secondTab).toHaveAttribute("aria-label", expect.stringContaining("State: Complete"));
+    });
+
+    it.each(["Business name", "Address line1", "Address city", "Address zip code", "Tax id", "Tax pin"])(
+      "renders tab two as incomplete if the text field %s is empty",
+      (emptyField) => {
+        renderComponent({});
+        const secondTab = screen.getAllByRole("tab")[1];
+        expect(secondTab).toHaveAttribute("aria-label", expect.stringContaining("State: Complete"));
+        fireEvent.click(secondTab);
+
+        fillText(emptyField, "");
+        expect(secondTab).toHaveAttribute("aria-label", expect.stringContaining("State: Incomplete"));
+      }
+    );
+
+    it("renders tab two as incomplete if Requesting Agency is not selected", async () => {
+      renderComponent({ business: generateBusinessWithEmptyTaxClearanceData() });
+      const secondTab = screen.getAllByRole("tab")[1];
+      expect(secondTab).toHaveAttribute("aria-label", expect.stringContaining("State: Incomplete"));
+      fireEvent.click(secondTab);
+
+      fillText("Business name", "Test Name");
+      fillText("Address line1", "123 Test Road");
+      fillText("Address city", "Baltimore");
+      selectDropdownValueByText("Address state", "MD", false);
+      fillText("Address zip code", "21210");
+      fillText("Tax id", "012345678901");
+      fillText("Tax pin", "1234");
+
+      expect(secondTab).toHaveAttribute("aria-label", expect.stringContaining("State: Incomplete"));
+
+      selectDropdownValueByText(
+        "Tax clearance certificate requesting agency",
+        LookupTaxClearanceCertificateAgenciesById("newJerseyBoardOfPublicUtilities").name,
+        true
+      );
+      expect(secondTab).toHaveAttribute("aria-label", expect.stringContaining("State: Complete"));
+    });
+
+    it("renders tab two as incomplete if Address state is not selected", async () => {
+      renderComponent({ business: generateBusinessWithEmptyTaxClearanceData() });
+      const secondTab = screen.getAllByRole("tab")[1];
+      expect(secondTab).toHaveAttribute("aria-label", expect.stringContaining("State: Incomplete"));
+      fireEvent.click(secondTab);
+
+      selectDropdownValueByText(
+        "Tax clearance certificate requesting agency",
+        LookupTaxClearanceCertificateAgenciesById("newJerseyBoardOfPublicUtilities").name,
+        true
+      );
+      fillText("Business name", "Test Name");
+      fillText("Address line1", "123 Test Road");
+      fillText("Address city", "Baltimore");
+      fillText("Address zip code", "21210");
+      fillText("Tax id", "012345678901");
+      fillText("Tax pin", "1234");
+
+      expect(secondTab).toHaveAttribute("aria-label", expect.stringContaining("State: Incomplete"));
```

Comment:
given how error handling works for the address component, the stepper should be in an error state